kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -198,9 +198,16 @@ public Optional<SetTransaction> getSetTxnOpt() {
         return setTxnOpt;
     }
 
+    private long generateInCommitTimestampForFirstCommitAttempt(long currentTimestamp) {
+        return currentTimestamp;
+    }
+
     private Row generateCommitAction() {
+        long commitAttemptStartTime = System.currentTimeMillis();
         return new CommitInfo(
-                System.currentTimeMillis(), /* timestamp */
+                generateInCommitTimestampForFirstCommitAttempt(
```

Comment:
Can you flag this change behind this table property: https://github.com/delta-io/delta/blob/59f8c64c2a2bc49877e8e19913c9249f9f337028/spark/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala#L765 ?